stacktrace.go

```diff
@@ -63,17 +55,15 @@ func takeStacktrace() string {
 	}
 
 	i := 0
-	skipZapFrames := true // skip all consecutive zap frames at the beginning.
```

Comment:
I'm not 100% sure what the impact of removing this is -- are all callers of `takeStacktrace` correctly passing skip so we no longer need this? Is there any scenario where it might not be correct?

Reply:
That's one of the things I tried changing the tests for. I already didn't use it when enabling this feature since the behavior with both skip and it was weird, I removed it now since it is not used with this being always enabled.

I explicitly changed the few places calling it so they pass a correct skip value. Also, Zap already took care to compensate for differences in call stack depth between `Logger` and `SugaredLogger` for the caller, so I reused that here.